packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -356,13 +366,35 @@ contract SystemConfig is OwnableUpgradeable, ISemver, IGasToken {
     /// @param _overhead New overhead value.
     /// @param _scalar   New scalar value.
     function _setGasConfig(uint256 _overhead, uint256 _scalar) internal {
+        require((uint256(0xff) << 248) & _scalar == 0, "SystemConfig: scalar exceeds max.");
```

Comment:
How do you feel about pulling out the `_scalar` decoding logic into internal helper methods? Arguably overkill since it's one line, but it lets us cleanly fuzz test the decoding (parsing this way vs. with foundry cheatcodes) to ensure it conforms to the spec

Reply:
This function still exists to support pre ecotone chains. There is no decoding here - this enforcement that the most significant byte is not set is just UX sugar. Its basically saying "use the `setGasConfig` interface if you are pre ecotone and use `setGasConfigEcotone` if you are post ecotone

Reply:
Ah perhaps you meant to leave this comment in the other function where it is encoded?

Reply:
For what its worth, there is e2e differential fuzzing of the solidity code that generates the scalar value, it doesn't directly operate over a library which you are right that is a thinner abstraction but it does directly fuzz over the abstraction that the `op-node` consumes

Reply:
Got it, that sounds sufficient to me